lib/fluent/plugin/out_mongo.rb

```diff
@@ -213,26 +219,33 @@ def collect_records(chunk)
         if date_keys
           date_keys.each { |date_key|
             begin
-              date_value = record[date_key]
+              date_key_accessor = @accessors[date_key.to_s]
+              date_value = date_key_accessor.call(record)
+              if date_value.to_i.to_s == date_value
+                date_value = date_value.to_i
+              elsif date_value.to_f.to_s == date_value
+                date_value = date_value.to_f
+              end
```

Comment:
Why this code added? Support `"123456.789"` like rare format?

Reply:
Yes, this is for legacy support. I have some source that use integer and float value casted as string and i need to re-cast it to the original type

Reply:
Could you move to String case with option?
It is very rare case and it should not be executed to avoid performance degradation.

```
else
  if @parse_string_number_date
    if date_value.to_i.to_s == date_value
      date_value = date_value.to_i
    elsif date_value.to_f.to_s == date_value
      date_value = date_value.to_f
    end
    value_to_set = date_value.is_a?(String) ? Time.parse(date_value) : Time.at(date_value)
  else
    value_to_set = Time.parse(date_value)
  end
end
```

Reply:
Implemented in latest commit